sdk/helper/automatedrotationutil/fields.go

```diff
@@ -75,6 +85,16 @@ func (p *AutomatedRotationParams) ShouldRegisterRotationJob() bool {
 	return p.RotationSchedule != "" || p.RotationPeriod != 0
 }
 
+func (p *AutomatedRotationParams) ShouldDeregisterRotationJob(previousConfig AutomatedRotationParams) bool {
+	return previousConfig.HasRotationJobConfigured() &&
+		(p.DisableAutomatedRotation ||
+			(p.RotationSchedule == "" && p.RotationPeriod == 0 && p.RotationWindow == 0))
+}
+
+func (p *AutomatedRotationParams) HasRotationJobConfigured() bool {
+	return !p.DisableAutomatedRotation && p.RotationSchedule != "" && p.RotationPeriod != 0 && p.RotationWindow != 0
```

Comment:
mutually exclusive; will never be set at the same time
also window check is not necessary due to being optional
```suggestion
	return !p.DisableAutomatedRotation && (p.RotationSchedule != "" || p.RotationPeriod != 0)
```